src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -812,8 +831,157 @@ def __set_errors_json(self, error_count_by_operation, errors_by_operation):
+            self.__installation_substatus_msg_copy = self.__get_substatus_message(truncated_status_file, installation_substatus_index)
+            self.__installation_packages_copy = self.__installation_substatus_msg_copy['patches']
+            low_pri_index = self.__get_installation_low_pri_index(self.__installation_packages_copy)
+
+        status_file_without_package_list_size = self.__size_of_constant_status_data(copy.deepcopy(truncated_status_file), assessment_substatus_index, installation_substatus_index)  # Deepcopy, fully copies the object to avoid reference modifications
+
+        size_of_max_packages_allowed_in_status = Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES - status_file_without_package_list_size
+        self.composite_logger.log_debug("Status file limits evaluated. [FileSizeInBytesWithoutPackages={0}] [MaxAllowablePackagesByteSize={1}]".format(
+            str(status_file_without_package_list_size), str(size_of_max_packages_allowed_in_status)))
+
+        while status_file_size_in_bytes > Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES:
+            # Start truncation process
+            packages_retained_in_assessment, packages_removed_from_assessment, packages_retained_in_installation, packages_removed_from_installation = \
+                self.__start_truncation_process(self.__assessment_packages_copy, self.__installation_packages_copy, size_of_max_packages_allowed_in_status, low_pri_index)
+
+            if len(packages_removed_from_assessment) > 0:
+                self.__assessment_packages_removed = packages_removed_from_assessment
+                self.composite_logger.log_debug("Recomposing truncated status payload: [Substatus={0}]".format(Constants.PATCH_ASSESSMENT_SUMMARY))
+                truncated_status_file = self.__recompose_truncated_status_file(truncated_status_file=truncated_status_file, truncated_package_list=packages_retained_in_assessment,
+                    substatus_message=self.__assessment_substatus_msg_copy, substatus_index=assessment_substatus_index)
+
+            if len(packages_removed_from_installation) > 0:
+                self.__installation_packages_removed = packages_removed_from_installation
+                self.composite_logger.log_debug("Recomposing truncated status payload: [Substatus={0}]".format(Constants.PATCH_INSTALLATION_SUMMARY))
+                truncated_status_file = self.__recompose_truncated_status_file(truncated_status_file=truncated_status_file, truncated_package_list=packages_retained_in_installation,
+                    substatus_message=self.__installation_substatus_msg_copy, substatus_index=installation_substatus_index)
+
+            status_file_size_in_bytes = self.__calc_status_size_on_disk(json.dumps(truncated_status_file))
+            status_file_agent_size_diff = status_file_size_in_bytes - Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES
```

Comment:
What does status_file_agent_size_diff refer to? Why is the word agent used here? IIUC, all our truncation work is in relation with an internal file size and not the agent file size

Reply:
i've renamed it to status_file_size_diff_in_bytes, pending for suggestions:


i can combine line 907 and 908 into one, but i wrote like that for readability that we need to calculate the extra byte size from tombstone, new errorrs, and escape characters, then subtract them from  max_allowed_patches_size_in_bytes.

i.e at first max_allowed_patches_size_in_bytes is 100kb,  after truncation the file is 100kb, but we add tombstone, truncated errors to maybe 5kb more, so now the file is 105kb, we need to keep it at 100kb, so max_allowed_patches_size_in_bytes should rlly be 95kb to account for extra 5kb. 

patches_retained_in_assessment, patches_removed_from_assessment, patches_retained_in_installation, patches_removed_from_installation = \
                self.__start_truncation_process(self.__assessment_patches_copy, self.__installation_patches_copy, max_allowed_patches_size_in_bytes, low_pri_index)

Reply:
pending

Reply:
This was already discussed before and had added in this PR multiple times. We need to maintain exclusivity between PRs. We cannot have code, which will only be used in future PRs, added here. Everything that merges in master should be functionally used.

We are not adding tombstone or error details in this PR, and hence no calculation on their sizes will take affect here. Why then do we need to add these lines which are going to be functional later?

Reply:
code is removed, done